addons/event_sale/tests/test_event_sale.py

```diff
@@ -376,7 +384,200 @@ def test_ticket_price_with_pricelist_and_tax(self):
             'event_id': event.id,
             'event_ticket_id': event_ticket.id,
         })
-        self.assertEqual(so.amount_total, 660.0, "Ticket is $1000 but the event product is on a pricelist 10 -> 6. So, $600 + a 10% tax.")
+        self.assertAlmostEqual(so.amount_total, 6.6, msg="Ticket is $1000 but the pricelist says fixed price $6 -> So, $6 + a 10% tax.")
+
+    def test_ticket_price_with_price_list_extensive(self):
+        """Extensive test of event ticket sales using price list.
+
+        Currently, multiple event ticket with various price can be linked to one product.
+        To easily integrate price list (which is based on product) features into event ticket, we override through
+        a context variable the product price during the computation of the price.
+        The goal of this test is to verify that the price is correctly computed using this mechanism using various
+        configuration of the price list.
+        Note that as this mechanism is duplicated in product and product template, we test both of them.
+        """
+        usd_currency = self.env['res.currency'].search([('name', '=', 'USD')])
+        eur_currency = self.env['res.currency'].with_context({'active_test': False}).search([('name', '=', 'EUR')])
+        company_currency = self.env.company.currency_id
+        other_currency = eur_currency if company_currency != eur_currency else usd_currency
+        precision_delta = other_currency.rounding
+        price_list = self.env['product.pricelist'].create({
+            'name': 'Sale pricelist',
+            'discount_policy': 'with_discount',
+        })
+        tax = self.env['account.tax'].create({
+            'name': "Tax 10",
+            'amount': 10,
+        })
+        event = self.env['event.event'].create({
+            'name': 'New Event',
+            'date_begin': '2020-02-02',
+            'date_end': '2020-04-04',
+        })
+        now = datetime.datetime.now()
+
+        def create_sale_order(is_product_templ,
+                              product_lst_price, ticket_price, qty,
+                              rule_type, rule_amount, min_qty,
+                              rule_based_on=None, product_standard_price=10,
+                              price_list_currency=None):
```

Comment:
Don't you always use `30.` as `product_lst_price` and `100.` as `ticket_price` in the tests? => Possible simplification of both the tests and this helper method.

Reply:
Yes, I have simplified the test following your advices.